File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        di = {}
        for ele in nums:
            if ele not in di:
                di[ele] = 1
        for i in range(1, len(nums)+1):
            if i not in di.keys():
                return i
        return len(nums)+1

File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        unordered_map<int, int> mp;
        for(int i: nums)
            mp[i]++;
        for(int i=1 ; i<=nums.size()+1 ; i++){
            if(mp[i]==0)
                return i;
        }
        return 1;
    }
};